packages/api/src/index.ts

```diff
@@ -57,25 +57,42 @@ passport.use(
     {
       clientID: env.githubClientId,
       clientSecret: env.githubSecret,
+      scope: 'user:emails',
       callbackURL: `${env.appUrl}/api/auth/github/callback`,
     },
     async (accessToken: any, refreshToken: any, profile: any, done: any) => {
       const currentUser = await authEm?.findOne(User, {
         githubId: profile.id,
       });
+
       if (!currentUser) {
-        logger.info('Creating new user.');
-        const newUser = new User({
-          name: profile.username,
-          githubId: profile.id,
-          hireable: false,
-          purpose: '',
+        const response = await fetch('https://api.github.com/user/emails', {
+          method: 'GET',
+          headers: {
+            Authorization: `token ${accessToken}`,
+          },
         });
-        await authEm?.persistAndFlush(newUser);
-        done(null, { profile, githubToken: accessToken });
+        const responseData = await response.json();
+
+        for (const emailResponse of responseData) {
+          // for loop to go through the emails and find one to match to, otherwise save null.
+          if (emailResponse.email.endsWith('.edu') && emailResponse.verified) {
+            logger.info('Creating new user.');
+            const newUser = new User({
+              name: profile.username,
+              githubId: profile.id,
+              hireable: false,
+              purpose: '',
+              email: emailResponse.email, // send the .edu email here
+            });
+            await authEm?.persistAndFlush(newUser);
+            return done(null, { profile, githubToken: accessToken });
+          }
+        }
```

Comment:
We need a `try catch` around anything that could throw unexpectedly and this has several `fetch` related calls that can throw: the `fetch` itself, `.json`, and the db stuff

In the `catch`, use the first argument of the `done` method to return a `new Error` with relevant error text and then send `null` as the second param